Userland/Libraries/LibGUI/Dialog.cpp

```diff
@@ -40,6 +41,20 @@ Dialog::ExecResult Dialog::exec()
             auto& parent_window = *static_cast<Window*>(parent());
             if (parent_window.is_visible()) {
                 window_rect.center_within(parent_window.rect());
+                if (window_rect.size().width() < desktop_rect.size().width() && window_rect.size().height() < desktop_rect.size().height()) {
+                    if (window_rect.top() < 20) {
```

Comment:
I'm actually not sure why this 20px offset is needed, if there is a constant defined somewhere I can reference for this I would appreciate a pointer

Reply:
That might be the height of the window title, I have a feeling that and the other borders aren't included in the window rect. You could test that by using a theme with thicker borders.

Reply:
Basically yes. The window frame is not part of its geometry, it's something that the WindowServer renders outside the window rect. :)